### ISSUE TITLE
`refund_bridge_transfer` should use resource account pattern or we should fund `origin_addr`

### ISSUE THREAD

**Issue by andygolay** [movementlabsxyz/movement#525]
**Describe the bug**
In `atomic_bridge_initiator` the `refund_bridge_transfer` function has some significant problems.

Notice first of all that there are two signers passed in. I fixed this for `complete_bridge_transfer` by implementing a resource account pattern via a helpful function `burn_moveth` in the counterparty Move module... but it's a little more complicated with `refund_bridge_transfer`.

```
    public fun refund_bridge_transfer(
        account: &signer,
        bridge_transfer_id: vector<u8>,
        atomic_bridge: &signer
    ) acquires BridgeTransferStore, BridgeConfig {
        assert!(signer::address_of(account) == @origin_addr, EINCORRECT_SIGNER);   
```        

Here it checks that `@origin_addr` must be the caller... is that correct? If so, that's fine, just need a way to call `dispatchable_fungible_asset::transfer(atomic_bridge, bridge_store, initiator_store, bridge_transfer.amount);` which is the main refund action.

```
        let config_address = borrow_global<BridgeConfig>(@atomic_bridge).bridge_module_deployer;
        let store = borrow_global_mut<BridgeTransferStore>(config_address);
        let bridge_transfer = aptos_std::smart_table::borrow_mut(&mut store.transfers, bridge_transfer_id);

        assert!(bridge_transfer.state == INITIALIZED, ENOT_INITIALIZED);
        assert!(timestamp::now_seconds() > bridge_transfer.time_lock, ENOT_EXPIRED);

        let originator_addr = bridge_transfer.originator;
        let bridge_addr = signer::address_of(atomic_bridge);
        let asset = moveth::metadata();

        // Transfer amount of asset from atomic bridge primary fungible store to originator's primary fungible store
        let initiator_store = primary_fungible_store::ensure_primary_store_exists(originator_addr, asset);
        let bridge_store = primary_fungible_store::ensure_primary_store_exists(@atomic_bridge, asset);
        dispatchable_fungible_asset::transfer(atomic_bridge, bridge_store, initiator_store, bridge_transfer.amount);
        
        ...
```    
I tried creating a helper utility function in `atomic_bridge_counterparty.move` like I did with `burn_moveth` but was having trouble passing in the correct addresses. 

Having all Move functions, for both initiator and counterparty, in one module would solve the resource account issue. Essentially I wasn't able to copy the resource signer cap or resource signer outside of the `atomic_bridge_counterparty.move` module. If anyone can show that it's possible, please do. I'm thinking it's likely a limitation of resource accounts as signers are not meant to be copied.

**To Reproduce**
1. View `protocol-units/bridge/move-modules/sources/atomic_bridge_initiator.move`
2. View `protocol-units/bridge/move-modules/sources/atomic_bridge_counterparty.move` 
3. Observe that it could work more elegantly to combine the initiator and counterparty functionality into one Move module 

Other solutions are welcome. It may be better to go for a short-term fix. I think I could get a helper function working for the refund transfer, but didn't want #523 to get too big so think it's better for its own PR.

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by andygolay** [comment]
Closing as I've made a fix for this in #523

**Comment by mzabaluev** [comment]
@andygolay It's better to keep this open, but close automatically as #523 gets merged. I will add this one to the list of issues resolved by that PR.
